src/curve.rs

```diff
@@ -902,58 +911,60 @@ impl ProjectivePoint {
     /// to the scalar.** If the scalar is fixed,
     /// this operation is effectively constant time.
     pub fn multiply_vartime(&self, by: &[u8; 32]) -> ProjectivePoint {
+        let digits = Scalar::bytes_to_wnaf_vartime(by, 5);
+
+        // We skip unset digits
+        let mut i: usize = 255;
+        for j in (0..256).rev() {
+            i = j;
+            if digits[i] != 0 {
+                break;
+            }
+        }
+        let table = NafLookupTable::<8>::from(self);
         let mut acc = ProjectivePoint::identity();
-        let table = LookupTable::<16>::from(self);
 
-        for digit in by
-            .iter()
-            .rev()
-            .flat_map(|byte| (0..2).rev().map(move |i| (byte >> (i * 4)) & 15u8))
-        {
-            acc = acc.double_multi(4);
-            if digit != 0 {
-                acc += table.get_point_vartime(digit as i8);
+        loop {
+            acc = acc.double();
+
+            match digits[i].cmp(&0) {
+                Ordering::Greater => acc += &table.get_point(digits[i] as usize),
+                Ordering::Less => acc -= &table.get_point(-digits[i] as usize),
+                Ordering::Equal => (),
+            };
+
+            if i == 0 {
+                break;
             }
+
+            i -= 1;
```

Comment:
Is it more efficient to do it this way than to have e.g. `for i = (0..i+1).rev()`?

Reply:
I believe it's the same, I'm afraid of readability here as `i` refers to a loop indexed by `i`, so I instead used `for j in (0..i + 1).rev()`.